marklogic-data-hub/src/main/java/com/marklogic/hub/hubcentral/HubCentralManager.java

```diff
@@ -160,9 +160,10 @@ protected void extractZipToProject(HubProject hubProject, File zipFile) {
                 outputFile.getParentFile().mkdirs();
                 try (InputStream inputStream = zip.getInputStream(entry);
                      FileOutputStream fileOut = new FileOutputStream(outputFile)) {
-                    inputStream.read(buffer, 0, entrySize);
-                    logger.info("Writing file: " + outputFile);
-                    FileCopyUtils.copy(buffer, fileOut);
+                    if (inputStream.read(buffer, 0, entrySize) > 0) {
```

Comment:
I think the fileOut declaration needs to be moved into the "if" block - otherwise, it never gets closed. Could just do:

    FileCopyUtils.copy(buffer, new FileOutputStream(outputFile));

Reply:
In the event when `FileCopyUtils.copy(buffer, fileOut)` does not run the fileOut stream should be closed via the try-with-resources block.